src/EditorFeatures/CSharpTest/AssignOutParameters/AssignOutParametersTests.cs

```diff
@@ -0,0 +1,534 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.AssignOutParameters;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AssignOutParameters
+{
+    public partial class AssignOutParametersTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new AssignOutParametersCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSimpleReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        [|return 'a';|]
+    }
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        return 'a';
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSwitchSectionReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        switch (0)
+        {
+            default:
+                [|return 'a';|]
+        }
+    }
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        switch (0)
+        {
+            default:
+                i = 0;
+                return 'a';
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingWhenVariableAssigned()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        [|return 'a';|]
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestWhenNotAssignedThroughAllPaths1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+            i = 1;
+        
+        [|return 'a';|]
+    }
+}",
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+            i = 1;
+        i = 0;
```

Comment:
Interesting... Wonder if the preferred pattern is to definitely assign it to default value at _start_ of the method, and preserve any subsequent assignments. So, for this test:

```csharp
    char M(bool b, out int i)
    {
        i = 0;
        if (b)
            i = 1;
        return 'a';
    }
}
```

Reply:
That sounds good for this case.

Reply:
We can probably offer that as another fix.   Woudl that be ok with you @mavasani ?  i.e. you'd get two actions `initialize out parameters` and `initialize out parameters (at start of method)`

Reply:
Sounds reasonable. Would you like to do it as part of this PR or a separate PR? Also how about offering the start of method fix also on the parameter definitions?

Reply:
> Sounds reasonable. Would you like to do it as part of this PR or a separate PR?

I'm happy to do that as part of this review.

> Also how about offering the start of method fix also on the parameter definitions?

I'll consider it.  That's more of a general refactoring, and it's less relevant to me than having a fix for someone who runs intoa  compiler error. :)

Reply:
added another codefix.